ClinGen/clincoded
# Connects to #322. Connects to #475. Connects to #613. Connects to #622. Connects to #629. Assessment Issues

**Changes for tickets #475 and #622**
  1. In family page, add a note to indicate that assessing a segregation is not allowed since no segregation data entered.
  2. Add a note to indicate that assessing family/segregation or experimental is not allowed since the evidence has been not assessed by its owner.
  3. Add existing assessment info in family, experimental and variant curation/edit and view pages.
Code change in family_curation.js, experimental_curation.js, variant_curation.js and assessment.js


**Change for ticket #629 (Bug of assessing family without segregation data)**
  1. Add control code in assessment.js to force assessment selection change to Not Assessed when all segregation data is deleted.
  2. Edit code to not count value "Not Assessed" as a valid assessment.


**Test of Family/Segregation Assessment**
* Create a new GDM and add a PMID.
* Generate a new family, expect to see a new note before enter segregation.data.
![screen shot 2016-03-14 at 12 39 04 pm](https://cloud.githubusercontent.com/assets/9206696/13757455/c6381520-e9e1-11e5-889d-c26ea43a3373.png)

* Enter any segregation field and expect to see:
![screen shot 2016-03-03 at 12 49 59 pm](https://cloud.githubusercontent.com/assets/9206696/13509052/7d9b3de8-e13e-11e5-8449-6a9ad7ac8a9f.png)

* Delete entered data, expect to see the note and disabled assessment selection again.

* Save the family without segregation data. Open view page and expect to see the note and assessment selection disabled as:
![screen shot 2016-03-14 at 12 37 00 pm](https://cloud.githubusercontent.com/assets/9206696/13757404/7ed3f38e-e9e1-11e5-890f-d2ddfbffef22.png)

* Logout and login as 2nd user. View the family and expect to see another note and disabled selection.
![screen shot 2016-03-04 at 12 15 59 pm](https://cloud.githubusercontent.com/assets/9206696/13540184/bba226f8-e209-11e5-960f-97f54bf22977.png)

* Logout and login as the 1st user. Edit family to enter segregation data  but not assess it and save. Logout and login as 2nd user. View the family and expect to see the same " ... not yet assessed on it." note and disabled selection.
![screen shot 2016-03-03 at 12 53 08 pm](https://cloud.githubusercontent.com/assets/9206696/13509153/ea44fe52-e13e-11e5-9bd3-99018f8e44f7.png)

* Logout and login as 1st user. Assess the family to Supports and save. Logout and login as the 2nd user and view the family. Expect to see the owner's assessment and to be able to assess the family.
![screen shot 2016-03-03 at 12 55 23 pm](https://cloud.githubusercontent.com/assets/9206696/13509229/48eeab6a-e13f-11e5-961b-7a993b52d9b7.png)

* Select Contradicts and save. Open the family view page again and expect to see all assessments info from both 1st and 2nd users.
![screen shot 2016-03-03 at 12 58 42 pm](https://cloud.githubusercontent.com/assets/9206696/13509324/ae6e2538-e13f-11e5-8eef-c02aa5e73230.png)


**Test of Experimental Assessment**
* Login as the 1st user.
* Create a new experimental, not assess it and save. Logout and login as 2nd user, view the experimental. Expect to see the "... not yet assessed on it." note and disabled selection.
![screen shot 2016-03-04 at 9 48 39 am](https://cloud.githubusercontent.com/assets/9206696/13535346/7e18ef12-e1ee-11e5-8e2b-22d3dc856ea8.png)

* Logout and login as the 1st user. Assess the experimental to Supports and save. Logout and login as 2nd user, view the experimental and expect to be able to assess.
![screen shot 2016-03-04 at 9 55 34 am](https://cloud.githubusercontent.com/assets/9206696/13535474/46321906-e1ef-11e5-95e4-78eba10bc587.png)

* Assess to Contradicts and save. Open the view page again and expect to see assessment info of both users.
![screen shot 2016-03-04 at 9 57 06 am](https://cloud.githubusercontent.com/assets/9206696/13535502/7891c978-e1ef-11e5-98aa-df86780eaa89.png)


**Test of Variant Assessment**
* Login as the 1st user.
* Associate a variant to the family. Curate the variant and assess it to Supports, save. Logout and login as 2nd user. Curate the variant and assess as Contradicts, save. Logout and login as 3rd user. Curate the variant and expect to see assessment info of both 1st and 2nd users.
![screen shot 2016-03-04 at 10 13 32 am](https://cloud.githubusercontent.com/assets/9206696/13535945/dd1f1ace-e1f1-11e5-92bd-bc46211c1fbf.png)

**Test of Ticket #629**
*  Bug of assessing family without segregation data
    * Curate a family, enter a segregation and select assessment to Supports.
    * Delete entered/selected data in all segregation fields. Expect to see Assessment selection value changing to Not Assessed.
  * Save and open edit family page, expect to see opened segregation input fields, Assessments: None, "... segregation data has been entered." note and Not Assessed selection.
![screen shot 2016-03-14 at 10 13 25 am](https://cloud.githubusercontent.com/assets/9206696/13752744/7099a642-e9cd-11e5-8ecc-c44e80663e89.png)
* Enter some segregation data and assess it to Review, save. Open family view page, expect to see:
![screen shot 2016-03-14 at 12 58 20 pm](https://cloud.githubusercontent.com/assets/9206696/13758020/75f8e226-e9e4-11e5-9c3c-21a714432404.png)

* Handling assessment Not Assessed
    * Assess any evidence and save.
    * Change to Not Assessed, expect to see None at Assessments panel.
![screen shot 2016-03-04 at 4 10 08 pm](https://cloud.githubusercontent.com/assets/9206696/13543789/9e5de874-e223-11e5-8e2c-2160c8e2986d.png)

**End of Test**
